test/integration/datatypes_test.exs

```diff
@@ -192,4 +192,55 @@ defmodule DataTypesTest do
     assert Map.fetch!(row, "map_of_int_to_text") == nil
     assert Map.fetch!(row, "set_of_int") == nil
   end
+
+  test "user defined types", %{conn: conn} do
+    statement = """
+    CREATE TYPE IF NOT EXISTS name (
+      first_name text,
+      last_name text,
+    )
+    """
+    Xandra.execute!(conn, statement)
+
+    statement = """
+    CREATE TYPE IF NOT EXISTS profile (
+      username text,
+      name frozen<name>,
+    );
+    """
+    Xandra.execute!(conn, statement)
+
+    statement = """
+    CREATE TABLE users (
+      id int PRIMARY KEY,
+      profile frozen<profile>,
+    );
+    """
+    Xandra.execute!(conn, statement)
+
+    statement = """
+    INSERT INTO users (id, profile) VALUES (?, ?)
+    """
+    foo_profile = %{
+      "username" => "foo",
+      "name" => %{"first_name" => "Kung", "last_name" => "Foo"},
+    }
+    bar_profile = %{
+      "username" => "bar",
+      "name" => %{"last_name" => "Bar"},
+    }
+    prepared = Xandra.prepare!(conn, statement)
+    Xandra.execute!(conn, prepared, [1, foo_profile])
+    Xandra.execute!(conn, prepared, [2, bar_profile])
+
+    statement = """
+    SELECT id, profile FROM users
+    """
+    page = Xandra.execute!(conn, statement)
+    assert [foo, bar] = Enum.to_list(page)
+    assert foo["id"] == 1
```

Comment:
We started using `Map.fetch!/2` instead of the access (`[]`) syntax here for the extra checks it gives for free.